### DISCUSSION TITLE
Make  jointjs  rectangle inside an other one and make graph interact with any other html component

### DISCUSSION THREAD

**Discussion by ybenarbiaBBI** [clientIO/joint#1847] [Q&A]
Hello, 

I want to know if it is possible to generate a jointjs Rectangle into an other rectangle  generated like this in a same graph.
I wanted to do something like this : 
![Capture](https://user-images.githubusercontent.com/99811254/196441625-f50ea724-3b37-43c6-8ddd-83a7a0931231.PNG)

Also i have a <p-table></p-table> (it's a primeng table component with some data) component inside the same page of my graph so i want to know if it could be possible for me to change style of each component of my jointJS graph by clicking on any row of my data table ?

for example : if i click  on a row, it would change color style of  rectangle and the date

Thank for your answers !

**Comment by jamesgeorgewilliams** [comment]
Yes, it's possible to create custom elements using JointJS. You can read how to create them in the following tutorial:
https://resources.jointjs.com/tutorial/custom-elements

To create an element like your picture, you could try something like the following:

```js
const namespace = joint.shapes;

const graph = new joint.dia.Graph({}, { cellNamespace: namespace });

const paper = new joint.dia.Paper({
    el: document.getElementById('paper'),
    model: graph,
    width: 600,
    height: 600,
    cellViewNamespace: namespace
});

const Element = joint.dia.Element.define('custom.Element', {
    attrs: {
        body: {
            width: 'calc(w)',
            height: 'calc(h)',
            strokeWidth: 'none',
            fill: 'lightgreen',
            rx: '15'
        },
        label: {
            textVerticalAnchor: 'middle',
            textAnchor: 'middle',
            x: 'calc(w/4)',
            y: 'calc(h/5)',
            fontSize: 14,
            fill: '#333333',
            text: 'Selection'
        },
        rectSecondary: {
            width: 'calc(w)',
            height: 'calc(h/2)',
            y: 'calc(h/2)',
            strokeWidth: 'none',
            fill: 'lightgrey',
            rx: '15'
        },
        secondaryLabel: {
            textVerticalAnchor: 'middle',
            textAnchor: 'middle',
            x: 'calc(w/2)',
            y: 'calc(h/1.4)',
            fontSize: 14,
            fill: '#333333',
            text: 'V1 / 13-07-18'
        },
    }
}, {
    markup: [{
        tagName: 'rect',
        selector: 'body',
    }, {
        tagName: 'text',
        selector: 'label'
    }, {
        tagName: 'rect',
        selector: 'rectSecondary',
    }, {
        tagName: 'text',
        selector: 'secondaryLabel'
    }]
});

Object.assign(namespace, {
    custom: {
        Element
    }
});

const myCustomElement = new Element({
    size: { width: 150, height: 80 }
});

myCustomElement.addTo(graph);
```

I am not familiar with primeng tables, but I certainly imagine it will be possible to do this.

Most likely, you will need to create some 'click' events listeners in the table which then change attributes in the corresponding cell in JointJS.

It's quite easy to update a cell's attributes in JointJS with something like `myCustomElement.attr({ body: { fill: 'lightblue' }});`.

**Reply by jamesgeorgewilliams** [reply]
You might also find some inspiration from the following demo:
https://resources.jointjs.com/demos/diagram-index
https://github.com/clientIO/joint-plus-tutorial-angular/tree/angular-material-tree

The component on the left is an Angular component. When you select one of the Items in the Angular component, it adds a highlight to the corresponding JointJS cell. You could update the shape color or text in a similar manner.
